client/js/components/StatsLeaderboard.vue

```diff
@@ -0,0 +1,359 @@
+<template>
+  <div>
+    <div id="stats-table-upper-surface" class="d-flex align-end">
+      <v-select
+        v-model="selectedMetric"
+        :items="metricChoices"
+        label="Select Metric"
+        class="filter-select mr-4 fit"
+        data-cy="metric-select"
+      />
+      <v-select
+        v-model="selectedWeeks"
+        :items="weeks"
+        label="Select Weeks"
+        data-cy="week-select"
+        class="flex-grow-1"
+        menu-props="data-week-select-menu"
+        multiple
+        chips
+        deletable-chips
+      />
+    </div>
+    <v-data-table
+      :items="tableRows"
+      :headers="tableColumns"
+      :sort-by.sync="sortBy"
+      :loading="loading"
+      :item-class="tableRowClass"
+    >
+      <template #[`item.rank`]="{item, value}">
+        <span :data-rank="item.username">{{ value }} </span>
+      </template>
+      <template v-for="week in ['total', ...selectedWeeks]" #[`item.week_${week}`]="{item, value}">
+        <v-tooltip :key="`${item.username}_week_${week}_wins`" top v-if="value">
+          <template #activator="{on, attrs}">
+            <v-chip
+              :key="`${item.username}_week_${week}`"
+              :color="colorForScoreByWeek(week, value)"
+              dark
+              :outlined="['primary', '#000'].includes(colorForScoreByWeek(week, value))"
+              v-bind="{
+                ...attrs,
+                ...dataAttribute(item.username, week),
```

Comment:
Data attribute is dynamically generated html attribute for selection during testing e.g. `week-1="player1"`

Reply:
Something I've done before that I find useful is to prefix any sort of test data attribute with `data-qa-*` so that I can look them up more easily in the future. Not anything you need to do here but food for thought since you aren't using it anywhere else in the code.

Reply:
Yeah I've been inconsistent about that. This is actually the only place in the code I think that doesn't use data-* as a prefix. I've inconsistently used data-cy-* sometimes and just data-* othertimes when the names get long, which could use cleaning. For now, there's no good reason not to at least use data-* here so I'll add that